sklearn/cluster/tests/test_bicluster.py

```diff
@@ -262,3 +262,13 @@ def test_n_features_in_(est):
     assert not hasattr(est, "n_features_in_")
     est.fit(X)
     assert est.n_features_in_ == 3
+
+
+@pytest.mark.parametrize("est", (SpectralBiclustering(), SpectralCoclustering()))
+def test_inf_values_handling_in_(est):
+    X = np.array([[1, 1, 2, 0, 0], [0, 1, 0, 1, 0], [0, 0, 1, 0, 0], [1, 2, 0, 2, 0]])
+
+    try:
+        est.fit(X)
+    except ValueError:
+        pytest.fail("test raised a ValueError because X contains infs or NaNs")
```

Comment:
Basically you don't need that. If `est` raises an error this is what we are going to catch.

Reply:
I also was convinced that first version of the test would work, but I had `Failed: DID NOT RAISE <class 'ValueError'>` after pushing https://github.com/scikit-learn/scikit-learn/pull/29988/commits/37f9a57d843860c149e606d215d8d7c475e7c412.  Logically it makes sense, since `fit` shouldn't raise an error when we have the mitigation added and working. How can I express this differently, or am I missing something?